k9mail-library/src/main/java/com/fsck/k9/mail/transport/smtp/SmtpTransport.java

```diff
@@ -660,76 +656,31 @@ private void writeLine(String s, boolean sensitive) throws IOException {
-            int replyCode = -1;
-            String message = line;
-            if (length >= 3) {
-                try {
-                    replyCode = Integer.parseInt(line.substring(0, 3));
-                } catch (NumberFormatException e) { /* ignore */ }
+    private static class CommandResponse {
 
-                if (length > 4) {
-                    message = line.substring(4);
-                } else {
-                    message = "";
-                }
-            }
+        private final int replyCode;
+        private final List<String> results;
 
-            throw new NegativeSmtpReplyException(replyCode, message);
+        public CommandResponse(int replyCode, List<String> results) {
+            this.replyCode = replyCode;
+            this.results = results;
         }
-
     }
-    @Deprecated
-    private List<String> executeSimpleCommand(String command) throws IOException, MessagingException {
-        return executeSimpleCommand(command, false);
-    }
-
-    /**
-     * TODO:  All responses should be checked to confirm that they start with a valid
-     * reply code, and that the reply code is appropriate for the command being executed.
-     * That means it should either be a 2xx code (generally) or a 3xx code in special cases
-     * (e.g., DATA & AUTH LOGIN commands).  Reply codes should be made available as part of
-     * the returned object.
-     *
-     * This should be done using the non-deprecated API below.
-     */
-    @Deprecated
-    private List<String> executeSimpleCommand(String command, boolean sensitive)
-    throws IOException, MessagingException {
-        List<String> results = new ArrayList<>();
-        if (command != null) {
-            writeLine(command, sensitive);
-        }
 
-        String line = readCommandResponseLine(results);
-
-        // Check if the reply code indicates an error.
-        checkLine(line);
-
-        return results;
+    private CommandResponse executeSensitiveCommand(String format, Object... args)
+            throws IOException, MessagingException {
+        return executeCommand(true, format, args);
     }
 
-    private static class CommandResponse {
-
-        private final int replyCode;
-        private final String message;
-
-        public CommandResponse(int replyCode, String message) {
-            this.replyCode = replyCode;
-            this.message = message;
-        }
+    private CommandResponse executeCommand(String format, Object... args) throws IOException, MessagingException {
+        return executeCommand(false, format, args);
     }
 
-    private CommandResponse executeSimpleCommandWithResponse(String command, boolean sensitive) throws IOException, MessagingException {
+    private CommandResponse executeCommand(boolean sensitive, String format, Object... args)
+            throws IOException, MessagingException {
         List<String> results = new ArrayList<>();
-        if (command != null) {
+        if (format != null) {
```

Comment:
`format` is the required format string. `args` contains the possibly empty arguments.